controllers/windowsmachine_controller.go

```diff
@@ -337,6 +332,24 @@ func (r *WindowsMachineReconciler) Reconcile(ctx context.Context, request ctrl.R
 	return ctrl.Result{}, nil
 }
 
+// deleteMachineIfAllowed deletes the specified Machine if deletion is allowed.
+// Returns a boolean if the deletion attempt should result in a requeuing.
```

Comment:
Consider leaving the `requeuing` decision out of the `deleteMachineIfAllowed()` func.

`deleteMachineIfAllowed()` returns whether the machine was deleted or not, then the requeuing logic takes place.

```go
   machineDeleted, err := r.deleteMachineIfAllowed(machine)

   //  error handling ...

   // requeue based on machineDeleted ...
```

Reply:
I like this suggestion. Thanks Jose